src/app/auth/HaalCentraalApi.ts

```diff
@@ -0,0 +1,51 @@
+import { ApiClient } from '@gemeentenijmegen/apiclient';
+import { Bsn, AWS } from '@gemeentenijmegen/utils';
+
+export class HaalCentraalApi {
+
+  private endpoint: string;
+  private client: ApiClient;
+
+  constructor(client: ApiClient) {
+    this.client = client;
+    if (!process.env.BRP_HAAL_CENTRAAL_API_URL) {
+      throw new Error('Could not initialize brp api as no endpoint parameter is provided in BRP_API_URL');
+    }
+    if (!process.env.BRP_API_KEY) {
+      throw new Error('Could not initialize brp api as no key parameter is provided in BRP_API_KEY');
+    }
+    this.endpoint = process.env.BRP_HAAL_CENTRAAL_API_URL;
+  }
+
+  /**
+   * Bevraag de Haal Centraal BRP API
+   * @param bsn the bsn to get data for
+   * @returns the API response data of {error: ""}
+   */
+  async getBrpData(bsn: string) {
+    try {
+      const apiKey = await AWS.getSecret(process.env.BRP_API_KEY_ARN!);
+      const aBsn = new Bsn(bsn);
+      let data = await this.client.postData(this.endpoint,
+        {
+          type: 'RaadpleegMetBurgerservicenummer',
+          fields: ['aNummer', 'adressering', 'burgerservicenummer', 'datumEersteInschrijvingGBA', 'datumInschrijvingInGemeente', 'europeesKiesrecht', 'geboorte', 'gemeenteVanInschrijving', 'geslacht', 'gezag', 'immigratie', 'indicatieCurateleRegister', 'indicatieGezagMinderjarige', 'kinderen', 'leeftijd', 'naam', 'nationaliteiten', 'ouders', 'overlijden', 'partners', 'uitsluitingKiesrecht', 'verblijfplaats', 'verblijfstitel', 'verblijfplaatsBinnenland', 'adresseringBinnenland'],
+          burgerservicenummer: aBsn.bsn,
+        },
+        {
+          'Content-type': 'application/json',
+          'X-API-KEY': apiKey,
+        });
+
+      if (data?.Persoon?.overleden) {
```

Comment:
Zo te zien bestaat dit niet, response zal iets als `data.personen[0]` bevatten, overlijden zo te zien onder `data?.personen[0]?.overlijden`

Reply:
Klopt deze check kan ook weg, want de bij de haal centraal api moet je expliciet aangeven dat je ook overleden wil hebben.

Reply:
Ik vond dat wat onduidelijk, bij de 'zoek'-opdrachten moet je 't meegeven, bij de `RaadpleegMetBurgerservicenummer` zie ik dat niet, en zie ik in de response wel de mogelijkheid voor een 'overlijden' key
![Uploading Screenshot 2024-07-25 at 16.22.47.png…]()

Reply:
<img width="920" alt="Screenshot 2024-07-25 at 16 29 41" src="https://github.com/user-attachments/assets/0f558728-286f-4696-a21b-4fef33d3a368">
Klopt, maar als je dan daaronder kijkt, zie je die mogelijke param wel bij bijv. `ZoekMetGeslachtsnaam(etc)`, maar niet bij de `RaadpleegMetBurgerservicenummer`, wat mij doet vermoeden dat je dan gewoon altijd die persoon krijgt, ook als ie overleden is. (Wat ook niet per se onlogisch is, als je al precies weet wie je wil hebben). Ik zou het of adhv de testset checken, of de check erin houden.

Reply:
Als je test met `999992077` (overleden persoon) als BSN zie je inderdaad een persoon met een 'overlijden' key terugkomen. Net in postman geprobeerd.

Reply:
Dan wordt het controleren adhv de testset, want in de response van de zoeken met bsn zit geen veld overleden.

Reply:
Ik denk dat onze comments elkaar hebben gekruist :D

Reply:
Ah ik zie het nu ook